src/problems/dde_problems.jl

```diff
@@ -43,3 +50,112 @@ DDEProblem(f, args...; kwargs...) =
 
 DDEProblem(f::AbstractDDEFunction, args...; kwargs...) =
   DDEProblem{isinplace(f)}(f, args...; kwargs...)
+
+"""
+$(TYPEDEF)
+"""
+abstract type AbstractDynamicalDDEProblem end
+
+"""
+$(TYPEDEF)
+"""
+struct DynamicalDDEProblem{iip} <: AbstractDynamicalDDEProblem end
+
+# u' = f1(v,h)
+# v' = f2(t,u,h)
+"""
+    DynamicalDDEProblem(f::DynamicalDDEFunction,v0,u0,tspan,p=NullParameters(),callback=CallbackSet())
+
+Define a dynamical DDE problem from a [`DynamicalDDEFunction`](@ref).
+"""
+function DynamicalDDEProblem(f::DynamicalDDEFunction,v0,u0,h::Function,tspan,p=NullParameters();dependent_lags=(),kwargs...)
+  DDEProblem(f,ArrayPartition(v0,u0),h,tspan,p;
+             problem_type=DynamicalDDEProblem{isinplace(f)}(),
+             dependent_lags=ntuple(i->(u,p,t)->dependent_lags[i](u[1],u[2],p,t),length(dependent_lags)),
+             kwargs...)
+end
+function DynamicalDDEProblem(f::DynamicalDDEFunction,h::Function,tspan,p=NullParameters();kwargs...)
+  DynamicalDDEProblem(f,h(p,first(tspan))...,h,tspan,p;kwargs...)
+end
+function DynamicalDDEProblem(f1::Function,f2::Function,args...;kwargs...)
```

Comment:
Varargs are used here so the optional `v0`, `u0` initial conditions can be handled by dispatch by the two constructors above, and the code doesn't have to be repeated for any of the following constructors. `f1` and `f2`'s types are specified to be `Function` so that the constructor doesn't just pull out the first 2 arguments of and try to pass on the rest, which would result in more confusing dispatch errors down the line if an incorrect call is made.